src/util/path.ts

```diff
@@ -2,13 +2,34 @@
 
 import { cache } from '../cache';
 import { config } from '../config';
-import { fabric } from '../../HEADER';
 import { halfPI, PiBy180 } from '../constants';
 import { commaWsp, rePathCommand } from '../parser/constants';
 import { Point } from '../point.class';
+import { PathData } from '../typedefs';
 import { cos } from './misc/cos';
-import { sin } from './misc/sin';
 import { multiplyTransformMatrices, transformPoint } from './misc/matrix';
+import { sin } from './misc/sin';
+
+type PathSegmentInfoCommon = {
+  x: number;
+  y: number;
+  command: string;
+  length: number;
+};
+
+type CurveInfo = PathSegmentInfoCommon & {
+  iterator: (pct: number) => Point;
+  angleFinder: (pct: number) => number;
+  length: number;
+};
+
+export type PathSegmentInfo = {
```

Comment:
we are not doing types with T anymore.
I think is good because it helps understand when is a type and when is class, on top of that i press `T` and i see 99% only types in the autocomplete.

At some point we ll have to sort this out

Reply:
Correct
I am not for and not entirely against.
The only true thing I dislike are name starting with T, then it is a bit weird: `TTransformEvent`

Reply:
Then names like `TEvent` might be confusing since we might have wanted to abbreviate `TransformEvent` into `TEvent`